wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/spi/InternalRuntime.java

```diff
@@ -111,32 +114,44 @@ public void start(Map<String, String> startOptions) throws InfrastructureExcepti
     /**
      * Starts underlying environment in implementation specific way.
      *
-     * @param startOptions options of workspace that may used in environment start
+     * @param startOptions
+     *         options of workspace that may used in environment start
      * @throws InternalInfrastructureException
      *         when error that indicates system internal problem occurs
+     * @throws RuntimeStartInterruptedException
+     *         when start execution is cancelled
      * @throws InfrastructureException
      *         when any other error occurs
      */
     protected abstract void internalStart(Map<String, String> startOptions) throws InfrastructureException;
 
     /**
-     * Stops Runtime
-     * Presumably can take some time so considered to launch in separate thread
+     * Stops Runtime.
+     * Presumably can take some time so considered to launch in separate thread.
+     *
+     * <p>Runtime will be stopped only if its state {@link WorkspaceStatus#RUNNING} or {@link WorkspaceStatus#STARTING}.
      *
-     * @param stopOptions  options of workspace that may used in environment stop
+     * <ul>
+     * <li>When runtime state is {@link WorkspaceStatus#STARTING} then process of start must be cancelled
```

Comment:
I think this contract should be described in javadocs of `internalStop` method because it specifies implementation behavior.